python/pyspark/ml/param/shared.py

```diff
@@ -311,7 +311,7 @@ class HasSeed(Params):
 
     def __init__(self):
         super(HasSeed, self).__init__()
-        self._setDefault(seed=hash(type(self).__name__))
+        self._setDefault(seed=None)
```

Comment:
Hm you may be right (looks like it converts to an Int actually? same point). I looked over the Python code, and virtually every class that extends HasSeed also appears to set seed=None by default. So in a way this change doesn't do much at all.

I'm going to correct the test I missed, and change one class that fixes an odd seed in its API, and propose that. It would at least change behavior on the Scala side, and make the Python side internally consistent and consistent with Scala.